queue_services/entity-emailer/src/entity_emailer/email_processors/agm_location_change_notification.py

```diff
@@ -0,0 +1,150 @@
+            'filingDateTime': filing_date_time,
+            'effectiveDateTime': effective_date if effective_date != filing_date_time else '',
+            'filingIdentifier': str(filing.id),
+            'businessNumber': business_data.tax_id if business_data and business_data.tax_id else ''
+        },
+        headers=headers
+    )
+    if receipt.status_code != HTTPStatus.CREATED:
+        logger.error('Failed to get receipt pdf for filing: %s', filing.id)
+    else:
+        receipt_encoded = base64.b64encode(receipt.content)
+        pdfs.append(
+            {
+                'fileName': 'Receipt.pdf',
+                'fileBytes': receipt_encoded.decode('utf-8'),
+                'fileUrl': '',
+                'attachOrder': attach_order
+            }
+        )
+        attach_order += 1
+
+    return pdfs
+
+
+def process(email_info: dict, token: str) -> dict:  # pylint: disable=too-many-locals, too-many-branches
+    """Build the email for AGM Location Change notification."""
+    logger.debug('agm_location_change_notification: %s', email_info)
+    # get template and fill in parts
+    filing_type, status = email_info['type'], email_info['option']
+    # get template vars from filing
+    filing, business, leg_tmz_filing_date, leg_tmz_effective_date = get_filing_info(email_info['filingId'])
+    filing_name = filing.filing_type[0].upper() + ' '.join(re.findall('[a-zA-Z][^A-Z]*', filing.filing_type[1:]))
+
+    template = Path(
+        f'{current_app.config.get("TEMPLATE_PATH")}/AGM-LOCCHG-{status}.html'
+    ).read_text()
+    filled_template = substitute_template_parts(template)
+    # render template with vars
+    jnja_template = Template(filled_template, autoescape=True)
+    filing_data = (filing.json)['filing'][f'{filing_type}']
+    html_out = jnja_template.render(
+        business=business,
+        filing=filing_data,
+        header=(filing.json)['filing']['header'],
+        filing_date_time=leg_tmz_filing_date,
+        effective_date_time=leg_tmz_effective_date,
+        entity_dashboard_url=current_app.config.get('DASHBOARD_URL') +
+        (filing.json)['filing']['business'].get('identifier', ''),
+        email_header=filing_name.upper(),
+        filing_type=filing_type
+    )
+
+    # get attachments
+    pdfs = _get_pdfs(token, business, filing, leg_tmz_filing_date, leg_tmz_effective_date)
+
+    # get recipients
+    identifier = filing.filing_json['filing']['business']['identifier']
+    recipients = []
+    recipients.append(get_recipient_from_auth(identifier, token))
+
+    recipients = list(set(recipients))
+    recipients = ', '.join(filter(None, recipients)).strip()
+
+    # assign subject
+    subject = 'AGM Location Change Documents from the Business Registry'
+
+    legal_name = business.get('legalName', None)
+    legal_name = 'Numbered Company' if legal_name.startswith(identifier) else legal_name
+    subject = f'{legal_name} - {subject}' if legal_name else subject
+
```

Comment:
I couldn't find a `is_numbered` check or equivalent, so I used an approach from [colin_api/filing.py L1274](https://github.com/bcgov/lear/blob/main/colin-api/src/colin_api/models/filing.py#L1274) to determine if a business is a numbered company.